app.rb

```diff
@@ -47,6 +46,21 @@ def log_comparison(comparison)
     puts line.to_json
   end
 
+  def log_level(comparison)
+    if (comparison[:different_keys].empty? || comparison[:different_keys] == "N/A") &&
+        matches?(comparison, :status) &&
+        matches?(comparison, :body_size)
+      :info
+    else
+      :warn
+    end
```

Comment:
Is this logic correct? I'm not sure if I follow...

Are we trying to say "If the comparison says that the keys are all the same, but the status and body_size are the same, then only log at `warn` level"?

If so, could we potentially be dropping warning where the keys are the same, but the responses are different (but the same size)?

Reply:
Oh, I get it. I thought "different_keys" meant "these objects have different keys", and that there would be another method called "different_values" that checked values. It's not that, it's "get a list of keys where there are different values". So this is saying everything's the same.

Reply:
`if (comparison[:different_keys].empty? || comparison[:different_keys] == "N/A") &&`
if (the keys are all the same, or it wasn't parseable JSON) 
AND 
the response statuses are the same
AND
the body sizes are the same

then log at info level

if any of those are not true, log at warn level

Reply:
The one case that this doesn't catch is when the responses are _not_ JSON, and the bodies are the same size and the same status, but there is a difference in at least one character of the body.

But I can't think of a case where that would occur, except for maybe an error - can you?